src/test/func/first_operation/first_operation.cc

```diff
@@ -194,18 +194,18 @@ int main(int, char**)
   printf("\n");
   for (size_t exp = 1; exp < snmalloc::SUPERSLAB_BITS; exp++)
   {
-    f(1ULL << exp);
-    f(3ULL << exp);
-    f(5ULL << exp);
-    f(7ULL << exp);
-    f((1ULL << exp) + 1);
-    f((3ULL << exp) + 1);
-    f((5ULL << exp) + 1);
-    f((7ULL << exp) + 1);
-    f((1ULL << exp) - 1);
-    f((3ULL << exp) - 1);
-    f((5ULL << exp) - 1);
-    f((7ULL << exp) - 1);
+    f(static_cast<size_t>(1) << exp);
```

Comment:
This makes me very sad.  Perhaps, define a local lambda to do this? Something like:
```C++
auto shifted = [exp](size_t v) { v << exp };
f(shifted(1));
f(shifted(3));
```
If that doesn't cause any warnings.